src/api/apiHelpers.native.test.ts

```diff
@@ -145,6 +166,63 @@ describe('[api] helpers', () => {
       expect(response).toEqual(await expectedResponse)
     })
 
+    it('should call refreshAccessToken route once when no error', async () => {
+      mockGetAccessTokenStatus.mockReturnValueOnce('expired').mockReturnValueOnce('expired')
+      const expectedResponse = respondWith('some api response')
+      mockFetch
+        .mockResolvedValueOnce(respondWith({ accessToken }))
+        .mockResolvedValueOnce(expectedResponse)
+        .mockResolvedValueOnce(expectedResponse)
+
+      await Promise.all([
+        safeFetch(apiUrl, optionsWithAccessToken, api),
+        safeFetch(apiUrl, optionsWithAccessToken, api),
+      ])
+
+      const refreshAccessTokenCalls = 1
+      const apiURLCalls = 2
+      expect(mockFetch).toHaveBeenCalledTimes(refreshAccessTokenCalls + apiURLCalls)
+    })
+
+    it("should not call refreshAccessToken route again before 15 minutes (access token's lifetime) have elapsed", async () => {
+      mockGetAccessTokenStatus.mockReturnValueOnce('expired').mockReturnValueOnce('expired')
+      const expectedResponse = respondWith('some api response')
+      mockFetch
+        .mockResolvedValueOnce(respondWith({ accessToken }))
+        .mockResolvedValueOnce(expectedResponse)
+        .mockResolvedValueOnce(expectedResponse)
+
+      await safeFetch(apiUrl, optionsWithAccessToken, api)
+
+      jest.advanceTimersByTime(15 * 60 * 1000 - 1)
+
+      await safeFetch(apiUrl, optionsWithAccessToken, api)
+
+      const refreshAccessTokenCalls = 2
+      const apiURLCalls = 1
```

Comment:
suggestion (non-blocking mais quand meme vivement recommandée)

```suggestion
      const refreshAccessTokenCalls = 1
      const apiURLCalls = 2
```

Reply:
Oups